flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionObserverTest.java

```diff
@@ -369,6 +371,42 @@ void testObserverWithEnableLossesCompensation() {
         report.acNodalInjections.forBaseCase().forEach((inj, p) -> assertFalse(inj.startsWith(LossesCompensator.LOSSES_ID_PREFIX)));
     }
 
+    @Test
+    void testPtdfsStayTheSameWithLossCompensationAndSlackDistributionOnLoads() {
+        String networkFileName = "ptdf_instability.xiidm";
+        Network network = TestUtils.importNetwork(networkFileName);
+        XnecProvider xnecProvider = new XnecProviderAllBranches();
+
+        LoadFlowParameters loadFlowParameters = new LoadFlowParameters();
+        loadFlowParameters.setBalanceType(LoadFlowParameters.BalanceType.PROPORTIONAL_TO_LOAD).setDistributedSlack(true);
+        FlowDecompositionParameters flowDecompositionParameters = new FlowDecompositionParameters().setEnableLossesCompensation(false);
+
+        // Without loss compensation
+        FlowDecompositionComputer flowDecompositionComputer1 = new FlowDecompositionComputer(flowDecompositionParameters, loadFlowParameters);
+        var report1 = new ObserverReport();
+        flowDecompositionComputer1.addObserver(report1);
+        flowDecompositionComputer1.run(xnecProvider, network);
+
+        // With loss compensations
+        flowDecompositionParameters.setEnableLossesCompensation(true);
+        FlowDecompositionComputer flowDecompositionComputer2 = new FlowDecompositionComputer(flowDecompositionParameters, loadFlowParameters);
+        var report2 = new ObserverReport();
+        flowDecompositionComputer2.addObserver(report2);
+        flowDecompositionComputer2.run(xnecProvider, network);
+
+        var ptdfs1 = report1.ptdfs.forBaseCase();
+        var ptdfs2 = report2.ptdfs.forBaseCase();
+
+        // Ensure that ptdf signs are the same with or without loss compensation
```

Comment:
Could you explain why you want to keep the same ptdf sign with or without loss compensation please ?

Reply:
Actually, I expect the same PTDF results with or without loss compensation.

In this case the PTDF changes sign (+1 becomes -1) and to avoid thresholding I thought it would be simpler to only compare signs.

Reply:
Yes, I understand your test but I'm wondering why does it matter to keep the same pdtf value before or after loss compensation ? 

From my point of view, loss compensation allows to to DC computations that are closer to AC computations without paying AC computation costs. So having slightly different PTDF between DC-loss-compensated PTDF and AC PTDF is expected.  

Futhermore, I don't understand why this test might improve sign stability ?

Reply:
Keeping the same PTDF sign matters to have similar (not the same due to losses) allocated flows in flow decomposition results.

Here, PTDFs are not sligthly different, they are as different as they can be (+1 and -1). I'm betting it's a sign issue somewhere.

My test doesn't improve sign stability, it's just a minimal example showing there is one.

I haven't found out why this happens...

Reply:
I have investigated your issue.

The sign difference of the PTDF comes from the fact that we [change the sign of the PTDF value (see SensitivityAnalyser line 93)](https://github.com/powsybl/powsybl-entsoe/blob/main/flow-decomposition/src/main/java/com/powsybl/flow_decomposition/SensitivityAnalyser.java#L93).

If we remove this sign modification, then there is no more issue with your test.
However, this sign modification is here to ensure that burdening flows are positive and relieving flows negative without having to consider the reference flow direction.
This sign modification is only here for easier understanding of the flow decomposition results.
[More details are available here](https://powsybl.readthedocs.io/projects/entsoe/en/latest/flow_decomposition/flow-decomposition-outputs.html#flow-sign-conventions)


Futhermore, it could be a great idea to refactor the SensitivityAnalyser line 93 to extract a static function with a good name such as `respectFlowSignConvention`
```
    static double respectFlowSignConvention(double ptdfValue, double referenceFlow) {
        return referenceFlow < 0 ? -ptdfValue : ptdfValue;
    }
```

With this in mind, I think the best choice is to acknowledge this sign convention and change your test.
```
assertEquals(respectFlowSignConvention(ptdfValue1, dcFlow1.get(branchId)), respectFlowSignConvention(ptdfValue2, dcFlow2.get(branchId)), 1E-9);
```
This solution uses the fact that `respectFlowSignConvention` is an [involution](https://en.wikipedia.org/wiki/Involution_(mathematics)) (maybe we should test that as well). 
Maybe we should document the sign convention in the PTDF observer part also.

Thanks to @phiedw and @wangjer for their help.

Thank you in advance,

Hugo

Reply:
Thank you @OpenSuze @phiedw and @wangjer for this detailed answer!

The behaviour is much clearer to me now :)

I have updated the test and the docs accordingly.